src/atomic/atomic_cell.rs

```diff
@@ -0,0 +1,695 @@
+use core::cell::UnsafeCell;
+use core::fmt;
+use core::mem;
+use core::ptr;
+use core::slice;
+use core::sync::atomic::{self, AtomicBool, Ordering};
+
+/// A thread-safe mutable memory location.
+///
+/// This type is equivalent to [`Cell`], except it can also be shared among multiple threads.
+///
+/// Operations on `AtomicCell`s use atomic instructions whenever possible, and synchronize using
```

Comment:
I'm just curious -- can we avoid global locks?  If so, do we prefer global locks to per-object locks?

I'd like to ask if we document a little bit more on this design choice.

Reply:
I don't think we can do that without sticking an `AtomicBool` next to absolutely every object, even if they don't use locking. Note that `std::atomic` in C++ is implemented in a similar manner.

In case an object does need locking, I'm not really sure whether we prefer global locks over per-object locks. It's a tradeoff between memory consumption and performance.